frontend/src/components/SolvedTable/SolvedTable.component.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useEffect, useState } from "react";
+import {
+  Table,
+  Thead,
+  Tbody,
+  Tr,
+  Th,
+  Td,
+  TableCaption,
+  Center,
+  TableContainer,
+} from "@chakra-ui/react";
+import { Paginator } from "../Paginator/Paginator.component";
+import { SolvedQuestion, SolvedQnFilter } from "../../models/SolvedQuestion.model"; // Import the SolvedQuestion model
+
+export type TableProp = {
+  userId: number;
+  pageSize?: number;
+};
+
+export const SolvedTable = (props: TableProp) => {
+  const { userId, pageSize = 10 } = props;
+  const [currentPage, setCurrentPage] = useState(1);
+  const [solvedQuestions, setSolvedQuestions] = useState<SolvedQuestion[]>([]); // Use SolvedQuestion type
+  const [isLoaded, setIsLoaded] = useState(false);
+
+  const onPageChange = (page: React.SetStateAction<number>) => {
+    setCurrentPage(page);
+  };
+
+  useEffect(() => {
+    const fetchSolvedQuestions = async (): Promise<void> => {
+      if (userId) {
+        try {
+          const response = await fetch(`http://localhost:8000/api/users/${userId}/questions`);
+          if (response.ok) {
+            const data = await response.json();
+            setSolvedQuestions(data);
+            setIsLoaded(true);
+          } else {
```

Comment:
same as below, define your fetch inside the `/src/api` folder

Reply:
changed this as well in [Added user.ts to fetch userId and SolvedQuestions and changes to hompage](https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g10/pull/43/commits/aa3c823c21fa81988a8977712842575ec7e8f19c)